src/morse/api.clj

```diff
@@ -10,22 +10,17 @@
 
 (defn get-updates
   "Receive updates from Bot via long-polling endpoint"
-  [token {:keys [limit offset timeout]}]
-  (let [url      (str base-url token "/getUpdates")
-        query    {:timeout (or timeout 1)
-                  :offset  (or offset 0)
-                  :limit   (or limit 100)}
-        response (http/get url {:as               :json
-                                :query-params     query
-                                :throw-exceptions false})
-        {:keys [status body]} response]
-    (if (< status 300)
-      (:result body)
-
-      (do
-        (log/error "Telegram returned" (:status response)
-                   "from /getUpdates:" (:body response))
-        ::error))))
+  ([token resp-handler err-handler {:keys [limit offset timeout]}]
```

Comment:
I don't like that his looses logic of handling and reporting errors and it leaks to polling namespace.
I would prefer to go with channels here instead of callbacks

Reply:
Totally agree, don't like this interfaces with two callbacks either but didn't think through very well who needs to be responsible for error handling in general and what's the best way to propagate the errors to the end user in case of polling.